FAQ.md

```diff
@@ -1176,6 +1177,18 @@ So to exclude a file from the compilation, you need to exclude and all **all** f
 
 There is no way now to indicate an `“include”` to a file outside the current folder in the `tsconfig.json` (tracked by [#1927](https://github.com/Microsoft/TypeScript/issues/1927)). You can achieve the same result by either: 1. Using a `“files”` list, or 2. Adding a `/// <reference path="..." />` directive in one of the files in your directory.
 
+### Why am I getting the `error TS5055: Cannot write file 'xxx.js' because it would overwrite input file.` when using JavaScript files?
+
+For a TypeScript file, the TypeScript compiler by default emits the generated JavaScript files in the same directory with the same base file name.
+Because the TypeScript files and emitted JavaScript files always have different file extensions, it is safe to do so.
+However, if you have set the `allowJs` compiler option to `true` and didn't set any emit output options (`outFile` and `outDir`), the compiler will try to emit JavaScript source files by the same rule, which will result in the emitted JavaScript file having the same file name with the source file. To avoid accidently overwriting your source file, the  compiler will issue this warning and block the emitting.
+
+There are multiple ways to solve this issue, though all of them involve configuring compiler options, therefore it is recommended that you have a `tsconfig.json` or `jsconfig.json` file in the project root to enable this. 
+If you don't want JavaScript files included in your project at all, simply set the `allowJs` option to `false`.
+You may not have set the `allowJs` option to `true` yourself. In some cases, the option is set by default, for example, when you are using a `jsconfig.json` file instead of a `tsconfig.json` file.
```

Comment:
Not sure about this part. the vscode users who use `jsconfig` never emit files, so they should be fine. the VS users do not use `jsconfig` so this may be confusing to them.